nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/CaptureBoxEvents.java

```diff
@@ -0,0 +1,270 @@
+    public void stopped() {
+        if (eventStream != null && eventStream.isStarted()) {
+            eventStream.stop();
+        }
+    }
+
+    @Override
+    public List<ConfigVerificationResult> verify(ProcessContext context, ComponentLog verificationLogger, Map<String, String> attributes) {
+
+        final List<ConfigVerificationResult> results = new ArrayList<>();
+        BoxClientService boxClientService = context.getProperty(BoxClientService.BOX_CLIENT_SERVICE).asControllerService(BoxClientService.class);
+        boxAPIConnection = boxClientService.getBoxApiConnection();
+
+        try {
+            boxAPIConnection.refresh();
+            results.add(new ConfigVerificationResult.Builder()
+                    .verificationStepName("Box API Connection")
+                    .outcome(Outcome.SUCCESSFUL)
+                    .explanation("Successfully validated Box connection")
+                    .build());
+        } catch (Exception e) {
+            getLogger().warn("Failed to verify configuration", e);
+            results.add(new ConfigVerificationResult.Builder()
+                    .verificationStepName("Box API Connection")
+                    .outcome(Outcome.FAILED)
+                    .explanation(String.format("Failed to validate Box connection: %s", e.getMessage()))
+                    .build());
+        }
+
+        return results;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        if (events.isEmpty()) {
+            context.yield();
+            return;
+        }
+
+        final FlowFile flowFile = session.create();
+        final List<BoxEvent> boxEvents = new ArrayList<>();
+        final int recordCount = events.drainTo(boxEvents);
+
+        try (final OutputStream out = session.write(flowFile)) {
+            final Writer writer = new OutputStreamWriter(out, StandardCharsets.UTF_8);
+            writer.write("[");
+            final Iterator<BoxEvent> iterator = boxEvents.iterator();
+            while (iterator.hasNext()) {
+                BoxEvent event = iterator.next();
+                JsonObject jsonEvent = toRecord(event);
+                jsonEvent.writeTo(writer);
+                if (iterator.hasNext()) {
+                    writer.write(",");
+                }
+            }
+            writer.write("]");
+            writer.flush();
+        } catch (Exception e) {
+            getLogger().error("Failed to write events to FlowFile; will re-queue events and try again", e);
+            boxEvents.forEach(events::offer);
+            session.remove(flowFile);
+            context.yield();
+            return;
+        }
+
+        session.putAttribute(flowFile, "record.count", String.valueOf(recordCount));
+        session.putAttribute(flowFile, CoreAttributes.MIME_TYPE.key(), "application/json");
+        session.transfer(flowFile, REL_SUCCESS);
+    }
+
+    private JsonObject toRecord(BoxEvent event) {
```

Comment:
Is there a reason for creating the new `JsonObject` as opposed to using the `getSourceJson()` method on `BoxEvent`?

Reply:
The `getSourceJson()` method only returns a subset of the data contained in the BoxEvent object. We actually use it in this `toRecord` method:
````java
json.add("source", Objects.requireNonNullElse(event.getSourceJSON(), Json.NULL));
````